### PR TITLE
Create exitStatus field for Mixpanel (SCP-4948)

### PR DESCRIPTION
**Pull request by michelle-vel**
Adds a field to Mixpanel reports that will have an exitStatus, meaning a 0 if all works as expected and a 1 or specific error code if there was a failure

### CODE REVIEW

File: app/models/ingest_job.rb [hunk 1]
```diff
@@ -716,7 +732,8 @@ def get_job_analytics
       trigger: trigger,
       jobStatus: failed? ? 'failed' : 'success',
       machineType: vm_info['machineType'],
-      bootDiskSizeGb: vm_info['bootDiskSizeGb']
+      bootDiskSizeGb: vm_info['bootDiskSizeGb'],
+      exitStatus: exit_code #takes integer value from exit_code method
```
Comment (eweitz) [inline app/models/ingest_job.rb:736]: This example would clarify for maintainers how `exitStatus` is useful beyond `jobStatus`:

```suggestion
      exitStatus: exit_code # integer exit code from PAPI, e.g. `137` for out of memory (OOM)
```

**Review by eweitz — APPROVED** [review]
Code looks good!

Could you create a Mixpanel report that uses this new property in our production analytics, link it from this PR, and add it to the "[Ingest - Single Cell Portal](https://mixpanel.com/project/2120588/view/19059/app/boards)" board?  Like report "[Ingest job status](https://mixpanel.com/project/2120588/view/19059/app/boards#id=734187&editor-card-id=%22report-11688551%22)", but break down by `exitStatus` rather than `jobStatus`.  Maybe even add a filter like "`exitStatus` does not equal `0`", as the motivating marginal value of this property is to report _why_ ingest (e.g. differential expression) jobs fail.

I was concerned to see no indication we'd verified this handles "137" errors -- e.g., it seems plausible that exit status code 137 could be thrown upstream in PAPI, but then get swallowed and presented as a `1`.  I searched around and found no concrete examples of how PAPI presents 137's.  However, the "BACKGROUND" section of #1572 talks in some detail about it, mostly allaying my concern.

So it'd be nice to at least manually test this handles "137" / out of memory (OOM) errors as we expect, as that seems like the chief value of this work.  Jean might have a suitable example.  But I suspect this handles that properly, so that's not a blocking concern for me.

**Review by ehanna4 — COMMENTED** [review]
It looks like the test failure is related to these changes, I think there need to be updates to the tests for `ingest_job.rb` 

Looking at the test failure I see:

> *** Rails test failures ***
> 
>   IngestJobTest#test_should_gather_job_properties_to_report_to_mixpanel:
>   NoMethodError: unmocked method :done?, expected one of [:error, :metadata]
> Exiting with code: 1
> Error: Process completed with exit code 1.

**Comment by bistline** [comment]
> I was concerned this may well not log "137" / out of memory (OOM) errors, which seems like the chief value of this work. E.g., it seems plausible that exit status code 137 could be thrown upstream in PAPI, but then get swallowed and presented as a `1`. 

@eweitz when I was helping out with this PR, I did some investigation regarding the `exit_code` for pipelines.  From what I could tell there is only one status reported, and it is done so at the metadata event level, _not_ the pipeline level.  This _should_ correctly report `137` when the ingest pipeline process runs out of memory.  We did correctly see `65` for non-convention metadata files when we tried to validate them against the Alexandria convention, so it holds that OOM will be reported correctly as well.

**Comment by eweitz** [comment]
Thanks Jon, that updates my position on a manual test for 137 from "nice but non-blocking" to "likely unworthwhile".

**Review by bistline — APPROVED** [review]
Worked with Michelle to fix the mock errors in the tests - looks good otherwise!

**Pull request merged**